protocol/x/clob/keeper/process_operations_stateful_validation.go

```diff
@@ -80,6 +80,25 @@ func (k Keeper) MustFetchOrderFromOrderId(
 	return order
 }
 
+// statefulValidateMatchTaker performs stateful validation on the taker order of a specified match.
+func (k Keeper) statefulValidateMatchTaker(
```

Comment:
Can we rename this to note it verifies that immediate execution taker orders aren't already partially filled?

Reply:
Sure, was leaving the function name generic for future extensibility but this is all the logic it holds for now so maybe I don't even need this funciton

Reply:
Yeah up to you. Figured we can change function names later once we add more validation